src/org/traccar/database/PermissionsManager.java

```diff
@@ -39,6 +40,7 @@
     private volatile Server server;
 
     private final Map<Long, User> users = new ConcurrentHashMap<>();
+    private final Map<Long, String> usersTokens = new HashMap<>();
```

Comment:
I think you should flip the parameters. Lookup by token seems to be more frequent operation than creating new token.